client/src/components/layout/ShowService.js

```diff
@@ -29,30 +34,68 @@ const ShowService = (props) => {
     getServiceAndReviews()
   }, [])
 
-  const reviews = service.reviews.map((review) => {
+  const reviewsMap = service.reviews.map((review) => {
     return(
       <ReviewTile
-        key={review.id}
         review={review}
       />
     )
   })
 
+  const postReview = async (newReview) =>{
+    try {
+      const response = await fetch(`/api/v1/services/${props.match.params.id}/reviews`, {
+        method:"POST",
+        headers: new Headers ({
+          "Content-Type" : "application/json"
+        }),
+        body: JSON.stringify(newReview),
+      });
+      if(!response.ok){
+        if(response.status === 422){
+          const responseBody = await response.json()
+          const newErrors = translateServerErrors(responseBody.errors)
+          debugger
+          setErrors(newErrors)
+        } else{
+          throw (new Error(`${response.status} ${response.statusText}`))
+        }
+      }
+      const responseBody = await response.json()
+      setErrors([])
+      setService({...service, reviews: [...service.reviews, responseBody.review]})
+      return true
+    } catch (error) {
+      console.error(`Error in fetch: ${error.message}`)
+    }
+  }
+
+  let reviewFormMessage = <h1>Sign in to Add New Review</h1>
```

Comment:
Nice! We could also leave a handy link for users to be able to click to sign in/up for easy navigation